nostr-client-plus/src/bin/aggregator.rs

```diff
@@ -296,62 +302,76 @@ async fn run() -> Result<()> {
                 continue;
             }
 
-            match aggr_book.entry(job_id.clone()) {
-                Entry::Occupied(mut entry) => {
-                    tracing::debug!("Another result for job {} found", job_id);
-                    let (workers, payload) = entry.get_mut();
-
-                    if *payload != new_payload {
-                        // ToDo: here the logic can be complicated. For example, how do we
-                        //  handle resolutions now? Do we need a separate book to handle
-                        //  controversies?
-                        tracing::error!("payloads for {} are different", job_id);
-                        RESULT_ERRORS.inc();
+            // Falling below are logics for classification
+            let raw_data_id = hex::encode(new_payload.header.raw_data_id.hash());
+
+            // First fetch all existing tasks
+            let saved_tasks: Result<String, RedisError> =
+                (&mut redis_con_task).get(raw_data_id.clone());
+            let mut tasks: Vec<AssignerTask> = match saved_tasks {
+                Ok(val) => {
+                    match from_str(val.as_str()) {
+                        Ok(val) => {
+                            val
+                        }
 
-                        // ToDo: Currently we handle disputes by deleting the entry altogether.
-                        //  If this was already at the second result, another one will come and
-                        //  linger in the book because no other matches are expected to come.
-                        //  Another approach needs to decided.
-                        entry.remove();
-                        PENDING_JOBS.set(aggr_book.len()as i64);
-                        continue;
+                        Err(err) => {
+                            tracing::debug!("Failed parse tasks, error: {}", err);
+                            Vec::new()
```

Comment:
If there is an error, you `continue`, not return an empty vec and let the code go on..